pkg/network/kuryr_bootstrap.go

```diff
@@ -0,0 +1,161 @@
+package network
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/routers"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/networks"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	"net"
+
+	"github.com/pkg/errors"
+
+	netv1 "github.com/openshift/cluster-network-operator/pkg/apis/networkoperator/v1"
+)
+
+func ensureNetwork(client *gophercloud.ServiceClient, name string) (string, error) {
+	netId, err := networks.IDFromName(client, name)
+	if err == nil {
+		return netId, nil
+	} else {
+		opts := networks.CreateOpts{
+			Name: name,
+		}
+		netObj, err := networks.Create(client, opts).Extract()
+		if err != nil {
+			return "", errors.Wrap(err, "failed to create network")
+		}
+		return netObj.ID, nil
+	}
+}
+
+func ensureSubnet(client *gophercloud.ServiceClient, name, netId,
+	cidr string, gatewayIp *string) (string, error) {
+	subnetId, err := subnets.IDFromName(client, name)
```

Comment:
Is it possible to ensure that the cidr and gateway of the existing subnet are correct? We should (eventually) handle subnet expansion, at least for ServiceNetwork.

Reply:
Oh yes, we might want to look it up based on CIDR instead, good point.